tests/unit/test_client.py

```diff
@@ -1114,7 +1129,7 @@ def test_list_blobs_w_all_arguments_and_user_project(self):
             blobs = list(iterator)
 
             self.assertEqual(blobs, [])
-            connection.api_request.assert_called_once_with(
+            connection.api_request.args[0].assert_called_once_with(
```

Comment:
This test seems to indicate that with telemetry the return type is different than previous. Is this the case with and without telemetry enabled, or is it possible the type of api_request could be different?

Reply:
Yeah, so with or without telemetry we will do 
`value.api_request = functools.partial(telemetry_wrapped_api_request, value.api_request)`
Its just that with telemetry enabled, `telemetry_wrapped_api_request` will actually do stuff, whereas if telemetry is not enable its just a wrapper that calls the underlying api_request. In either case we need to call `args[0]` on the partial to get to the underlying api_request.

Reply:
As we add type-hints I wonder if this will surface in an odd way.

Reply:
Another way of accomplishing this could be to mock Connection.api_request, and assert that it gets called, instead of relying on the exact structure of the wrapper object, but then you'd probably also want to assert that the underlying meter got touched.